deepchem/models/torch_models/layers.py

```diff
@@ -47,3 +76,427 @@ def __init__(self, scale, eps=1e-5):
   def forward(self, x):
     norm = self.scale / torch.norm(x, dim=-1, keepdim=True).clamp(min=self.eps)
     return x * norm
+
+
+class Encoder(nn.Module):
+  """Encoder block for the Molecule Attention Transformer in [1]_.
+  
+  A stack of N layers which form the encoder block. The block primarily consists of a self-attention layer and a feed-forward layer.
+  This block is constructed from its basic layer: EncoderLayer. See dc.models.torch_models.layers.EncoderLayer for more details regarding the working of the block.
+
+  References
+  ----------
+  .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
+  
+  Examples
+  --------
+  >>> import deepchem as dc
+  >>> attention = dc.models.torch_models.layers.MATAttention('softmax', 0.33, 0.33)
+  >>> self_attn_layer = dc.models.torch_models.layers.MultiHeadedAttention(8, 1024, 0.1, attention)
+  >>> feed_fwd_layer = dc.models.torch_models.layers.PositionwiseFeedForward(d_input = 1024, activation = torch.nn.ReLU(), n_layers = 1, dropout = 0.1)
+  >>> block = dc.models.torch_models.layers.Encoder(self_attn_layer = self_attn_layer, feed_fwd_layer = feed_fwd_layer, d_model = 1024, dropout = 0.0, N = 3)
+  """
+
+  def __init__(self, self_attn_layer, feed_fwd_layer, d_model, dropout, N):
+    """Initialize an Encoder block.
+
+    Parameters
+    ----------
+    self_attn_layer: dc.torch_models.layers or nn.Module
+      Self-Attention layer to be used in the encoder block.
+    feed_fwd_layer: dc.torch_models.layers or nn.Module
+      Feed-Forward layer to be used in the encoder block.
+    d_model: int
+      Size of dense layer.
+    dropout: float
+      Dropout probability.
+    N: int
+      Number of identical layers to be stacked.
+    """
+
+    super(Encoder, self).__init__()
+    layer = EncoderLayer(
+        self_attn_layer=self_attn_layer,
+        feed_fwd_layer=feed_fwd_layer,
+        d_model=d_model,
+        dropout=dropout)
+    self.layers = clones(layer, N)
```

Comment:
Not sure we need the `clones` function here. You can just do
```
self.layers = nn.ModuleList([EncoderLayer(*args) for _ in range(N)])
```

Reply:
Done!